### ISSUE TITLE
flushEvents cause Swift App crash

### ISSUE THREAD

**Issue by LumenAnnex** [awslabs/clickstream-swift#23]
### Describe the bug

app crash by chance, stack shows it crash in `flushEvents`. 

### Steps To Reproduce

```swift
.
```

### Expected behavior

do not crash

### Clickstream-Swift Version

0.4.2

### Swift version

5.8

### Xcode version

14.3.1

### Relevant log output

```shell
Crashed: com.apple.root.background-qos
0  libswiftCore.dylib             0x402ca4 swift_unknownObjectRetain + 20
1  libswiftNetwork.dylib          0x391b0 initializeWithCopy for NWPath + 1292
2  libswiftNetwork.dylib          0x3ab60 outlined init with copy of NWPath + 88
3  libswiftNetwork.dylib          0x36ad0 NWPathMonitor.currentPath.getter + 88
4  gomerry                        0x236ed4 AWSClickstreamPlugin.flushEvents() + 22 (NetworkMonitor.swift:22)
5  gomerry                        0x239898 closure #2 in AWSClickstreamPlugin.configure(using:) + 60 (AWSClickstreamPlugin+Configure.swift:60)
6  gomerry                        0x257348 thunk for @escaping @callee_guaranteed () -> () + 4312625992 (<compiler-generated>:4312625992)
7  libdispatch.dylib              0x3eac _dispatch_client_callout + 20
8  libdispatch.dylib              0x7330 _dispatch_continuation_pop + 504
9  libdispatch.dylib              0x1a908 _dispatch_source_invoke + 1588
10 libdispatch.dylib              0x15944 _dispatch_root_queue_drain + 396
11 libdispatch.dylib              0x16158 _dispatch_worker_thread2 + 164
12 libsystem_pthread.dylib        0xda0 _pthread_wqthread + 228
13 libsystem_pthread.dylib        0xb7c start_wqthread + 8
```

### Is this a regression?

No

### Device

iPhone 11

### iOS Version

16.5.1

**Comment by zxkane** [comment]
Could you upgrade to 0.5.1 with a try?

**Comment by zhu-xiaowei** [comment]
This is a bug for concurrent read the `NWPathMonitor.currentPath` in some accident , we will fix it in the next version soon.

**Comment by LumenAnnex** [comment]
many thanks, does the version 0.5.1 solve the problem? If not, maybe I'll wait for the next version.

**Comment by zhu-xiaowei** [comment]
This issue will be fixed in 0.5.2, we need some time to test it, the release day should be within next week.

**Comment by zhu-xiaowei** [comment]
Released version 0.5.2 and fixed this bug, please upgrade with a try.

**Comment by LumenAnnex** [comment]
Many thanks! We will try in our next version.
